From: Johannes Thumshirn <jthumshirn@suse.de>
Date: Thu, 17 Dec 2015 12:37:20 +0100
Subject: nvme: Clear BIO_SEG_VALID flag in nvme_bio_split()
References: bsc#954992
Patch-Mainline: Never, SUSE specific

Clear BIO_SEG_VALID flag in nvme_bio_split() so no invalid length 0 bios are
submitted which causes NULL pointer dereferences.

This is a left over from non-backported bio split code.

Signed-off-by: Johannes Thumshirn <jthumshirn@suse.de>
---
 drivers/block/nvme-core.c |    3 +++
 1 file changed, 3 insertions(+)

--- a/drivers/block/nvme-core.c
+++ b/drivers/block/nvme-core.c
@@ -658,6 +658,9 @@ static struct nvme_bio_pair *nvme_bio_sp
 	bp->b1.bi_end_io = nvme_bio_pair_endio;
 	bp->b2.bi_end_io = nvme_bio_pair_endio;
 
+	bp->b1.bi_flags &= ~(1 << BIO_SEG_VALID);
+	bp->b2.bi_flags &= ~(1 << BIO_SEG_VALID);
+
 	bp->parent = bio;
 	atomic_set(&bp->cnt, 2);
 
